src/main/java/pvs/app/controller/GitlabApiController.java

```diff
@@ -0,0 +1,119 @@
+package pvs.app.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.gitlab4j.api.GitLabApiException;
+import pvs.app.dto.GitlabCommitDTO;
+import pvs.app.dto.GitlabIssueDTO;
+import pvs.app.service.GitlabApiService;
+import pvs.app.service.GitlabCommitService;
+
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+@RestController
+@RequestMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+public class GitlabApiController {
+
+    static final Logger logger = LogManager.getLogger(GithubApiController.class.getName());
+    private final GitlabApiService gitlabApiService;
+    private final GitlabCommitService gitlabCommitService;
+    @Value("${message.exception}")
+    private String exceptionMessage;
+
+    public GitlabApiController(GitlabApiService gitlabApiService, GitlabCommitService gitlabCommitService){
+        this.gitlabApiService = gitlabApiService;
+        this.gitlabCommitService = gitlabCommitService;
+    }
+
+    @SneakyThrows
+    @GetMapping("/gitlab/commits/{repoOwner}/{repoName}")
+    public ResponseEntity<String> getCommits(@PathVariable("repoOwner") String repoOwner, @PathVariable("repoName") String repoName) throws GitLabApiException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        List<GitlabCommitDTO> gitlabCommitDTOs = gitlabCommitService.getAllCommits(repoOwner, repoName);
+        String gitlabCommitDTOsJson;
+
+        try {
+            gitlabCommitDTOsJson = objectMapper.writeValueAsString(gitlabCommitDTOs);
+            return ResponseEntity.status(HttpStatus.OK)
+                    .body(gitlabCommitDTOsJson);
+        } catch (JsonProcessingException e) {
+            logger.debug(e.getMessage());
+            e.printStackTrace();
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(exceptionMessage);
+        }
+    }
+
+    @PostMapping("/gitlab/{username}/{password}")
+    public ResponseEntity<String> oauth2login(@PathVariable("username") String username, @PathVariable("password") String password) throws GitLabApiException {
+        System.out.println("going to execute gitlab api..");
+        this.gitlabApiService.oauth2login(username, password);
+        return ResponseEntity.status(HttpStatus.OK).body("log in succeed");
+    }
+
+    @PostMapping("/gitlab/commits/{repoOwner}/{repoName}")
+    public ResponseEntity<String> getCommitsFromGitlab(@PathVariable("repoOwner") String repoOwner, @PathVariable("repoName") String repoName) {
```

Comment:
This seems to be a duplicate or an unnecessary API.

The reason we need to establish an API here is because the front end needs to get this data, but there's already have the `getCommits` method, so we may not need to leave the actual interaction with gitlab to the front end to decide.

Therefore, it is recommended to merge the behavior of actually obtaining data into `getCommits`, because for the front-end, we do not need to expose the process of actually obtaining data to it. Therefore, the front-end should only call the back-end `getCommits`, and the back-end should be able to directly transfer the data. After packaging, send it back to the front end.